internal/services/resource/resource_group_resource.go

```diff
@@ -222,6 +226,9 @@ func resourceResourceGroupDelete(d *pluginsdk.ResourceData, meta interface{}) er
 
 	deleteFuture, err := client.Delete(ctx, id.ResourceGroup, "")
 	if err != nil {
+		if strings.Contains(err.Error(), "could not be found") {
```

Comment:
Sorry should have left a comment here as well, could apply the same check on the Response here like we did above?